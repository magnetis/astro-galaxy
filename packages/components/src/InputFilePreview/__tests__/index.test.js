import React from 'react';
import {
  InputFilePreview,
  InputFilePreviewItem,
  InputFilePreviewRemoveButton,
  InputFilePreviewText,
  InputFilePreviewImage,
  InputFilePreviewCloseIcon,
} from '@magnetis/astro-galaxy-components';

describe('InputFilePreview', () => {
  it('should render with image', () => {
    const json = rendererCreateWithTheme(
      <InputFilePreview>
        <InputFilePreviewItem>
          <InputFilePreviewRemoveButton>
            <InputFilePreviewCloseIcon />
          </InputFilePreviewRemoveButton>
          <InputFilePreviewImage
            alt="Logo Magnetis"
            src="https://i.ytimg.com/vi/t07h052QLG8/maxresdefault.jpg"
          />
        </InputFilePreviewItem>
      </InputFilePreview>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render with text', () => {
    const json = rendererCreateWithTheme(
      <InputFilePreview>
        <InputFilePreviewItem>
          <InputFilePreviewRemoveButton>
            <InputFilePreviewCloseIcon />
          </InputFilePreviewRemoveButton>
          <InputFilePreviewText>financialreport.csv</InputFilePreviewText>
        </InputFilePreviewItem>
      </InputFilePreview>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render with trucated text', () => {
    const json = rendererCreateWithTheme(
      <InputFilePreview>
        <InputFilePreviewItem>
          <InputFilePreviewRemoveButton>
            <InputFilePreviewCloseIcon />
          </InputFilePreviewRemoveButton>
          <InputFilePreviewText truncate>financialreport.csv</InputFilePreviewText>
        </InputFilePreviewItem>
      </InputFilePreview>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render mutiples items with image and text', () => {
    const json = rendererCreateWithTheme(
      <InputFilePreview>
        <InputFilePreviewItem>
          <InputFilePreviewRemoveButton>
            <InputFilePreviewCloseIcon />
          </InputFilePreviewRemoveButton>
          <InputFilePreviewImage
            alt="Logo Magnetis"
            src="https://i.ytimg.com/vi/t07h052QLG8/maxresdefault.jpg"
          />
        </InputFilePreviewItem>
        <InputFilePreviewItem>
          <InputFilePreviewRemoveButton>
            <InputFilePreviewCloseIcon />
          </InputFilePreviewRemoveButton>
          <InputFilePreviewText truncate>financialreport.csv</InputFilePreviewText>
        </InputFilePreviewItem>
        <InputFilePreviewItem>
          <InputFilePreviewRemoveButton>
            <InputFilePreviewCloseIcon />
          </InputFilePreviewRemoveButton>
          <InputFilePreviewText>financialreport.csv</InputFilePreviewText>
        </InputFilePreviewItem>
      </InputFilePreview>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });
});

describe('InputFilePreviewText', () => {
  it('should render trucated text with correct style', () => {
    const json = rendererCreateWithTheme(
      <InputFilePreviewText truncate>financialreport.csv</InputFilePreviewText>
    ).toJSON();

    expect(json).toHaveStyleRule('overflow', 'hidden');
    expect(json).toHaveStyleRule('text-overflow', 'ellipsis');
    expect(json).toHaveStyleRule('white-space', 'nowrap');
    expect(json).toMatchSnapshot();
  });

  it('should render full text with correct style', () => {
    const json = rendererCreateWithTheme(
      <InputFilePreviewText>financialreport.csv</InputFilePreviewText>
    ).toJSON();

    expect(json).toHaveStyleRule('overflow', 'auto');
    expect(json).toHaveStyleRule('text-overflow', 'clip');
    expect(json).toHaveStyleRule('white-space', 'normal');
    expect(json).toMatchSnapshot();
  });
});
