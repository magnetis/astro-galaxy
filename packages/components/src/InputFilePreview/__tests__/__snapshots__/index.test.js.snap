// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFilePreview should render mutiples items with image and text 1`] = `
.c3 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80px;
  height: 80px;
  margin-right: 32px;
  vertical-align: middle;
  background-color: #f6f7f8;
  border-radius: 4px;
  -webkit-box-pack: middle;
  -webkit-justify-content: middle;
  -ms-flex-pack: middle;
  justify-content: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  position: absolute;
  top: -6px;
  right: -6px;
  width: 20px;
  height: 20px;
  padding: 0;
  font-size: 12px;
  cursor: pointer;
  border: 0;
  border-radius: 50%;
  outline: none;
  background-color: #f6f7f8;
}

.c2:focus,
.c2:hover {
  background-color: #afbec9;
}

.c7 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  width: 80px;
  padding: 12px;
  margin: 0;
  font-size: 10px;
  word-wrap: break-word;
  overflow: auto;
  text-overflow: clip;
  white-space: normal;
}

.c6 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  width: 80px;
  padding: 12px;
  margin: 0;
  font-size: 10px;
  word-wrap: break-word;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c5 {
  width: 100%;
  height: 100%;
  border-radius: 4px;
  object-fit: cover;
}

.c4 {
  width: 20px;
  height: 20px;
  margin-left: -1px;
  color: #597183;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2"
    >
      <svg
        className="c3 c4"
        color="moon900"
        height={24}
        viewBox="0 0 32 32"
        width={24}
      >
        <defs>
          <path
            d="M16 14.586l4.793-4.793a1 1 0 0 1 1.414 1.414L17.414 16l4.793 4.793a1 1 0 0 1-1.414 1.414L16 17.414l-4.793 4.793a1 1 0 1 1-1.414-1.414L14.586 16l-4.793-4.793a1 1 0 0 1 1.414-1.414L16 14.586zM28.5 16c0 6.904-5.596 12.5-12.5 12.5S3.5 22.904 3.5 16 9.096 3.5 16 3.5 28.5 9.096 28.5 16zm-2 0c0-5.799-4.701-10.5-10.5-10.5S5.5 10.201 5.5 16 10.201 26.5 16 26.5 26.5 21.799 26.5 16z"
            id="icon-x-men-a"
          />
        </defs>
        <g
          id="icon/secondary/circle/close"
        >
          <mask
            id="icon-x-men-b"
          >
            <use
              xlinkHref="#icon-x-men-a"
            />
          </mask>
          <use
            id="Mask"
            xlinkHref="#icon-x-men-a"
          />
          <g
            id="color/moon-900"
            mask="url(#icon-x-men-b)"
          >
            <path
              d="M0 0h32v32H0z"
              id="Rectangle"
            />
          </g>
        </g>
      </svg>
    </button>
    <img
      alt="Logo Magnetis"
      className="c5"
      src="https://i.ytimg.com/vi/t07h052QLG8/maxresdefault.jpg"
    />
  </div>
  <div
    className="c1"
  >
    <button
      className="c2"
    >
      <svg
        className="c3 c4"
        color="moon900"
        height={24}
        viewBox="0 0 32 32"
        width={24}
      >
        <defs>
          <path
            d="M16 14.586l4.793-4.793a1 1 0 0 1 1.414 1.414L17.414 16l4.793 4.793a1 1 0 0 1-1.414 1.414L16 17.414l-4.793 4.793a1 1 0 1 1-1.414-1.414L14.586 16l-4.793-4.793a1 1 0 0 1 1.414-1.414L16 14.586zM28.5 16c0 6.904-5.596 12.5-12.5 12.5S3.5 22.904 3.5 16 9.096 3.5 16 3.5 28.5 9.096 28.5 16zm-2 0c0-5.799-4.701-10.5-10.5-10.5S5.5 10.201 5.5 16 10.201 26.5 16 26.5 26.5 21.799 26.5 16z"
            id="icon-x-men-a"
          />
        </defs>
        <g
          id="icon/secondary/circle/close"
        >
          <mask
            id="icon-x-men-b"
          >
            <use
              xlinkHref="#icon-x-men-a"
            />
          </mask>
          <use
            id="Mask"
            xlinkHref="#icon-x-men-a"
          />
          <g
            id="color/moon-900"
            mask="url(#icon-x-men-b)"
          >
            <path
              d="M0 0h32v32H0z"
              id="Rectangle"
            />
          </g>
        </g>
      </svg>
    </button>
    <p
      className="c6"
    >
      financialreport.csv
    </p>
  </div>
  <div
    className="c1"
  >
    <button
      className="c2"
    >
      <svg
        className="c3 c4"
        color="moon900"
        height={24}
        viewBox="0 0 32 32"
        width={24}
      >
        <defs>
          <path
            d="M16 14.586l4.793-4.793a1 1 0 0 1 1.414 1.414L17.414 16l4.793 4.793a1 1 0 0 1-1.414 1.414L16 17.414l-4.793 4.793a1 1 0 1 1-1.414-1.414L14.586 16l-4.793-4.793a1 1 0 0 1 1.414-1.414L16 14.586zM28.5 16c0 6.904-5.596 12.5-12.5 12.5S3.5 22.904 3.5 16 9.096 3.5 16 3.5 28.5 9.096 28.5 16zm-2 0c0-5.799-4.701-10.5-10.5-10.5S5.5 10.201 5.5 16 10.201 26.5 16 26.5 26.5 21.799 26.5 16z"
            id="icon-x-men-a"
          />
        </defs>
        <g
          id="icon/secondary/circle/close"
        >
          <mask
            id="icon-x-men-b"
          >
            <use
              xlinkHref="#icon-x-men-a"
            />
          </mask>
          <use
            id="Mask"
            xlinkHref="#icon-x-men-a"
          />
          <g
            id="color/moon-900"
            mask="url(#icon-x-men-b)"
          >
            <path
              d="M0 0h32v32H0z"
              id="Rectangle"
            />
          </g>
        </g>
      </svg>
    </button>
    <p
      className="c7"
    >
      financialreport.csv
    </p>
  </div>
</div>
`;

exports[`InputFilePreview should render with image 1`] = `
.c3 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80px;
  height: 80px;
  margin-right: 32px;
  vertical-align: middle;
  background-color: #f6f7f8;
  border-radius: 4px;
  -webkit-box-pack: middle;
  -webkit-justify-content: middle;
  -ms-flex-pack: middle;
  justify-content: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  position: absolute;
  top: -6px;
  right: -6px;
  width: 20px;
  height: 20px;
  padding: 0;
  font-size: 12px;
  cursor: pointer;
  border: 0;
  border-radius: 50%;
  outline: none;
  background-color: #f6f7f8;
}

.c2:focus,
.c2:hover {
  background-color: #afbec9;
}

.c5 {
  width: 100%;
  height: 100%;
  border-radius: 4px;
  object-fit: cover;
}

.c4 {
  width: 20px;
  height: 20px;
  margin-left: -1px;
  color: #597183;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2"
    >
      <svg
        className="c3 c4"
        color="moon900"
        height={24}
        viewBox="0 0 32 32"
        width={24}
      >
        <defs>
          <path
            d="M16 14.586l4.793-4.793a1 1 0 0 1 1.414 1.414L17.414 16l4.793 4.793a1 1 0 0 1-1.414 1.414L16 17.414l-4.793 4.793a1 1 0 1 1-1.414-1.414L14.586 16l-4.793-4.793a1 1 0 0 1 1.414-1.414L16 14.586zM28.5 16c0 6.904-5.596 12.5-12.5 12.5S3.5 22.904 3.5 16 9.096 3.5 16 3.5 28.5 9.096 28.5 16zm-2 0c0-5.799-4.701-10.5-10.5-10.5S5.5 10.201 5.5 16 10.201 26.5 16 26.5 26.5 21.799 26.5 16z"
            id="icon-x-men-a"
          />
        </defs>
        <g
          id="icon/secondary/circle/close"
        >
          <mask
            id="icon-x-men-b"
          >
            <use
              xlinkHref="#icon-x-men-a"
            />
          </mask>
          <use
            id="Mask"
            xlinkHref="#icon-x-men-a"
          />
          <g
            id="color/moon-900"
            mask="url(#icon-x-men-b)"
          >
            <path
              d="M0 0h32v32H0z"
              id="Rectangle"
            />
          </g>
        </g>
      </svg>
    </button>
    <img
      alt="Logo Magnetis"
      className="c5"
      src="https://i.ytimg.com/vi/t07h052QLG8/maxresdefault.jpg"
    />
  </div>
</div>
`;

exports[`InputFilePreview should render with text 1`] = `
.c3 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80px;
  height: 80px;
  margin-right: 32px;
  vertical-align: middle;
  background-color: #f6f7f8;
  border-radius: 4px;
  -webkit-box-pack: middle;
  -webkit-justify-content: middle;
  -ms-flex-pack: middle;
  justify-content: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  position: absolute;
  top: -6px;
  right: -6px;
  width: 20px;
  height: 20px;
  padding: 0;
  font-size: 12px;
  cursor: pointer;
  border: 0;
  border-radius: 50%;
  outline: none;
  background-color: #f6f7f8;
}

.c2:focus,
.c2:hover {
  background-color: #afbec9;
}

.c5 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  width: 80px;
  padding: 12px;
  margin: 0;
  font-size: 10px;
  word-wrap: break-word;
  overflow: auto;
  text-overflow: clip;
  white-space: normal;
}

.c4 {
  width: 20px;
  height: 20px;
  margin-left: -1px;
  color: #597183;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2"
    >
      <svg
        className="c3 c4"
        color="moon900"
        height={24}
        viewBox="0 0 32 32"
        width={24}
      >
        <defs>
          <path
            d="M16 14.586l4.793-4.793a1 1 0 0 1 1.414 1.414L17.414 16l4.793 4.793a1 1 0 0 1-1.414 1.414L16 17.414l-4.793 4.793a1 1 0 1 1-1.414-1.414L14.586 16l-4.793-4.793a1 1 0 0 1 1.414-1.414L16 14.586zM28.5 16c0 6.904-5.596 12.5-12.5 12.5S3.5 22.904 3.5 16 9.096 3.5 16 3.5 28.5 9.096 28.5 16zm-2 0c0-5.799-4.701-10.5-10.5-10.5S5.5 10.201 5.5 16 10.201 26.5 16 26.5 26.5 21.799 26.5 16z"
            id="icon-x-men-a"
          />
        </defs>
        <g
          id="icon/secondary/circle/close"
        >
          <mask
            id="icon-x-men-b"
          >
            <use
              xlinkHref="#icon-x-men-a"
            />
          </mask>
          <use
            id="Mask"
            xlinkHref="#icon-x-men-a"
          />
          <g
            id="color/moon-900"
            mask="url(#icon-x-men-b)"
          >
            <path
              d="M0 0h32v32H0z"
              id="Rectangle"
            />
          </g>
        </g>
      </svg>
    </button>
    <p
      className="c5"
    >
      financialreport.csv
    </p>
  </div>
</div>
`;

exports[`InputFilePreview should render with trucated text 1`] = `
.c3 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80px;
  height: 80px;
  margin-right: 32px;
  vertical-align: middle;
  background-color: #f6f7f8;
  border-radius: 4px;
  -webkit-box-pack: middle;
  -webkit-justify-content: middle;
  -ms-flex-pack: middle;
  justify-content: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  position: absolute;
  top: -6px;
  right: -6px;
  width: 20px;
  height: 20px;
  padding: 0;
  font-size: 12px;
  cursor: pointer;
  border: 0;
  border-radius: 50%;
  outline: none;
  background-color: #f6f7f8;
}

.c2:focus,
.c2:hover {
  background-color: #afbec9;
}

.c5 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  width: 80px;
  padding: 12px;
  margin: 0;
  font-size: 10px;
  word-wrap: break-word;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c4 {
  width: 20px;
  height: 20px;
  margin-left: -1px;
  color: #597183;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2"
    >
      <svg
        className="c3 c4"
        color="moon900"
        height={24}
        viewBox="0 0 32 32"
        width={24}
      >
        <defs>
          <path
            d="M16 14.586l4.793-4.793a1 1 0 0 1 1.414 1.414L17.414 16l4.793 4.793a1 1 0 0 1-1.414 1.414L16 17.414l-4.793 4.793a1 1 0 1 1-1.414-1.414L14.586 16l-4.793-4.793a1 1 0 0 1 1.414-1.414L16 14.586zM28.5 16c0 6.904-5.596 12.5-12.5 12.5S3.5 22.904 3.5 16 9.096 3.5 16 3.5 28.5 9.096 28.5 16zm-2 0c0-5.799-4.701-10.5-10.5-10.5S5.5 10.201 5.5 16 10.201 26.5 16 26.5 26.5 21.799 26.5 16z"
            id="icon-x-men-a"
          />
        </defs>
        <g
          id="icon/secondary/circle/close"
        >
          <mask
            id="icon-x-men-b"
          >
            <use
              xlinkHref="#icon-x-men-a"
            />
          </mask>
          <use
            id="Mask"
            xlinkHref="#icon-x-men-a"
          />
          <g
            id="color/moon-900"
            mask="url(#icon-x-men-b)"
          >
            <path
              d="M0 0h32v32H0z"
              id="Rectangle"
            />
          </g>
        </g>
      </svg>
    </button>
    <p
      className="c5"
    >
      financialreport.csv
    </p>
  </div>
</div>
`;

exports[`InputFilePreviewText should render full text with correct style 1`] = `
.c0 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  width: 80px;
  padding: 12px;
  margin: 0;
  font-size: 10px;
  word-wrap: break-word;
  overflow: auto;
  text-overflow: clip;
  white-space: normal;
}

<p
  className="c0"
>
  financialreport.csv
</p>
`;

exports[`InputFilePreviewText should render trucated text with correct style 1`] = `
.c0 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  width: 80px;
  padding: 12px;
  margin: 0;
  font-size: 10px;
  word-wrap: break-word;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<p
  className="c0"
>
  financialreport.csv
</p>
`;
